src/core/core.register.js

```diff
@@ -0,0 +1,45 @@
+import defaults from './core.defaults';
+
+/**
+ * @typedef {{id: string, defaults: any}} IChartComponent
+ */
+
+export default class Register {
```

Comment:
I might call this something like `TypedRegistry`. `Register` sounds funny to me since it's a verb

Reply:
done